File: Week Assignment 3 using C/10_print_elements.c
// 10. Hackerrank: Print elements of linked list
#include <stdio.h>
#include <stdlib.h>

struct Node { int data; struct Node* next; };

void display(struct Node* head) {
    while (head) {
        printf("%d\n", head->data);
        head = head->next;
    }
}

int main() {
    struct Node* head = (struct Node*)malloc(sizeof(struct Node)); head->data=1; head->next=NULL;
    head->next = (struct Node*)malloc(sizeof(struct Node)); head->next->data=2; head->next->next=NULL;
    head->next->next = (struct Node*)malloc(sizeof(struct Node)); head->next->next->data=3; head->next->next->next=NULL;
    display(head);
    return 0;
}

File: Week Assignment 3 using Python/10_print_elements.py
# 10. Hackerrank: Print elements of linked list
class Node:
    def __init__(self,data):
        self.data=data; self.next=None

def display(head):
    t = head
    while t:
        print(t.data)
        t = t.next

if __name__ == "__main__":
    head = Node(1); head.next = Node(2); head.next.next = Node(3)
    display(head)
